net: wifi: Add reasons for 4 way handshake failure

kapbh opened the pull request:
When there is 4 way handshake timeout, it can be because of incorrect credential or some times when AP's signal strength is weak. It can cause 4 way handshake fail. Adding comment where WIFI_STATUS_CONN_WRONG_PASSWORD is defined.

krish2718 commented on include/zephyr/net/wifi_mgmt.h:
```diff
@@ -367,6 +367,10 @@ struct wifi_connect_req_params {
 
 /** Wi-Fi connect result codes. To be overlaid on top of \ref wifi_status
  * in the connect result event for detailed status.
+ * The Possible reasons for 4 way handshake failure are as follows.
```
Shouldn't this note be for `WIFI_STATUS_CONN_WRONG_PASSWORD`? Where we can add that this is just a best guess.

kapbh replied:
Yes it's for WIFI_STATUS_CONN_WRONG_PASSWORD, I will move note to that declaration.

kapbh committed:
net: wifi: Add reasons for 4 way handshake failure

When there is 4 way handshake timeout, it can be because of
incorrect credential or some times when AP's signal strength
is weak. It can cause 4 way handshake fail. Adding comment
where WIFI_STATUS_CONN_WRONG_PASSWORD is defined.

Signed-off-by: Kapil Bhatt <kapil.bhatt@nordicsemi.no>
